app/controllers/api/v1/subjects_controller.rb

```diff
@@ -11,9 +11,12 @@ class Api::V1::SubjectsController < Api::ApiController
   def index
     case params[:sort]
     when 'cellect'
-      render json_api: SubjectSerializer.page(params, selector.cellect_subjects)
+      render json_api: SubjectSerializer.page(params,
+                                              selector.cellect_subjects,
+                                              selector_context)
```

Comment:
Just had a thought re avoiding paying the cost with context methods. Can we move the `selector_context` into the selector object and switch it depending on whether cellect returned results? It's be nice not to check if the workflow is finished / a user has finished each time we want subjects.